### DISCUSSION TITLE
Can't get FindControl to find my control when using TemplatedControl

### DISCUSSION THREAD

**Discussion by donnyv** [AvaloniaUI/Avalonia#7692] [Q&A]
I built a control using template control following Nick Polyak tutorial. Which works and displays fine when I run it.  
https://www.codeproject.com/Articles/5317055/Multiplatform-Avalonia-NET-Framework-Programming-A

I want to add a default tab to the AuraTabView which is is in the template control xaml code. When the control is first created. I keep getting the error "System.InvalidOperationException: 'Could not find parent name "

For some reason it can't find the name "TabView". Even though thats what the "Name" property is set too. I also noticed that there are no controls in the VisualChildren property. Do I need to call this code from some where else after the tree has bee built? Not sure what to do.

```
public class TabsContainerControl : TemplatedControl
    {
        public TabsContainerControl()
        {
            var t = this.VisualChildren;

            var tabview = this.FindControl<AuraTabView>("TabView");

        }
}
```

**Comment by robloo** [comment]
At the time the constructor is run, the control template has not been applied. This means the system hasn't even created the parts of the UI that represent the control.

Standard practice is to use the OnApplyTemplate overridable method for your FindControl code. Then store a reference for use everywhere else. Be careful to always null check before use -- it's also standard practice to not crash even if no template is applied.

**Reply by donnyv** [reply]
I don't get the error anymore. But can't seem to use FindControl with TemplatedControl
![image](https://user-images.githubusercontent.com/2224189/155829002-ae148852-a933-4c63-8a03-388fcb7775a1.png)

**Reply by maxkatz6** [reply]
Use ```e.Namescope.Find```

Template was just applied, control wasn't yet added to the visual tree, so there is no point to check it.

**Reply by robloo** [reply]
Example:

https://github.com/AvaloniaUI/Avalonia/blob/cf40772dd1bb9651887387c9d687f209d3c210e1/src/Avalonia.Controls/SplitButton/SplitButton.cs#L233
